core/state_transition.go

```diff
@@ -405,6 +418,20 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	}
 	st.gasRemaining -= gas
 
+	if rules.IsEIP4762 {
+		targetAddr := msg.To
+		originAddr := msg.From
+
+		st.evm.Accesses.AddTxOrigin(originAddr.Bytes())
+
+		if msg.To != nil {
+			st.evm.Accesses.AddTxDestination(targetAddr.Bytes(), msg.Value.Sign() != 0)
```

Comment:
Why is this needed? We're going to `Call` into that address, shouldn't that add it later on?

Reply:
because when it comes to witnesses, these costs are paid for the 21000 gas. All other calls are paid for. If this is moved to the `Call`, I have to add an `if` somewhere to distinguish between the top-level call, and any other, nested call.